### PR TITLE
[Grid] Refactor grid-areas

### PR DESCRIPTION
**Pull request by kyledurand**
This really cleans up the Grid component and improves performance by removing resize event listener

### CODE REVIEW

File: polaris-react/scripts/gen-media-queries.js [hunk 1]
```diff
@@ -59,5 +61,5 @@ function getDownMediaCondition(breakpoint) {
 }
 
 function toEm(value) {
-  return `${parseInt(value, 10) / 16}em`;
+  return `${parseInt(value, 10)}em`;
```
Comment (kyledurand) [inline polaris-react/scripts/gen-media-queries.js]: I need to revert this change once https://github.com/Shopify/polaris/pull/6012 ships

File: polaris-react/src/components/Grid/Grid.scss [hunk 2]
```diff
@@ -25,6 +35,16 @@
...
+    grid-template-areas: var(
+      --pc-grid-areas-xl,
+      var(
+        --pc-grid-areas-lg,
+        var(
+          --pc-grid-areas-md,
+          var(--pc-grid-areas-sm, var(--pc-grid-areas-xs))
+        )
+      )
+    );
```
Comment (kyledurand) [inline polaris-react/src/components/Grid/Grid.scss]: cc @aaronccasanova we have all the fallbacks here now (which we couldn't have had with the js solution without a closest match check) by chaining vars with fallbacks. Code looks grosssss but super nice to use

Comment (aaronccasanova) [reply polaris-react/src/components/Grid/Grid.scss]: I love the idea!! Had a quick play with a similar/alternative approach: https://codepen.io/aaronccasanova/pen/QWQxpmE?editors=1100

What do you think? In the context of the `Grid`, I'm imagining we would set the base case to:
`--pc-grid-areas-xs: initial`

**Review by alex-page — APPROVED** [review]

**Review by aaronccasanova — APPROVED** [review]

File: polaris-react/src/components/Grid/Grid.scss (hunk 2 above)
Comment (kyledurand) [reply polaris-react/src/components/Grid/Grid.scss]: I missed this til just now but this is a fantastic idea. Will implement
